libs/stac/odc/stac/_eo3.py

```diff
@@ -83,19 +84,17 @@ def asset_geobox(asset: pystac.asset.Asset) -> GeoBox:
     :raises ValueError: when transform is not Affine.
     """
     _proj = ProjectionExtension.ext(asset)
-    assert _proj.shape is not None
-    assert _proj.transform is not None
-    assert _proj.crs_string is not None
+    if _proj.shape is None or _proj.transform is None or _proj.crs_string is None:
+        raise ValueError("Missing proj data")
```

Comment:
```suggestion
        raise ValueError("The asset must have the following fields (from the projection extension): shape, transform, and one of an epsg, wkt2, or projjson")
```

Also, does this code path support a projection extension defined on the `Item` rather than the `Asset`?

Reply:
My assumption was that item level proj data is exposed on an asset automatically by `pystac`, but I have not verified that, I just assumed that if this works for crs it works for others, but maybe not cause crs is special and always shared between assets?